docs/guides/10-add-legends.md

```diff
@@ -236,7 +222,70 @@ With that change, the final map will now label other categories as "Other weathe
         frameBorder="0">
     </iframe>
 </div>
-You can explore the final map [here](/developers/carto-vl/examples/maps/guides/add-legends/step-5.html)
+You can explore this step [here](/developers/carto-vl/examples/maps/guides/add-legends/step-5.html)
+
+### Assign opacity through variables
+
+If you want to assign opacity to mapped features, you will construct your `ramp` using [variables](/developers/carto-vl/guides/Glossary/#variables). 
+
+**Note:**
+The use of [variables](/developers/carto-vl/guides/Glossary/#variables) is explored in greater detail in the [Add Widgets](/developers/carto-vl/guides/add-widgets/) guide.
+
+The styling below assigns a global `0.5` opacity to features, using two variables: 
+* `@myRamp: ramp($weather, [darkorange, darkviolet, darkturquoise])` for the colors assigned to each category 
+* `@myOpacity: 0.5` for the amount of opacity to assign to features 
+
+The variables are then used in the `color` property inside of an `opacity` expression: 
+
+```CARTO_VL_Viz
+const viz = new carto.Viz(`
+    @myRamp: ramp($weather, [darkorange, darkviolet, darkturquoise])
+    @myOpacity: 0.5
+
+    width: 7
+    color: opacity(@myRamp,@myOpacity)
+    strokeWidth: 0.2
+    strokeColor: black
+`);
+```
+
+Next, you need to modify where the legend gets the data to draw (`getLegendData()`). In the previous examples, we set the data to come from the `color` property but since the symbology is assigned through variables in this case, we need to modify where it is read from in `const colorLegend`:
+
+```html
+// Request data for legend from the layer variable myRamp
+            const colorLegend = layer.viz.variables.myRamp.getLegendData();
+            let colorLegendList = '';
+```
+At this point, if you refresh your map, you will see that the features on the map have the set opacity, but the features in the legend do not.
+
+If you want to add opacity to the legend items, you need to remove the `function rgbToHex(color)`, add constants for opacity (`const alpha`) and color (`const rgba`) where you can reference `alpha` for the `a` value of `const rgba`. Finally, you need to set the legend list items to draw from `const rgba`: 
```

Comment:
IMO having so many details talking about `consts` and `alpha` in this paragraph looks a bit confusing, more than the code itself. 

Maybe we could say something like: 
*"If you want to add opacity to the legend items, you need to remove the `function rgbToHex(color)` (working just with the RGB color components of each entry) and use another approach, which includes the transparency. In this case, we have included that as a viz variable, named `myOpacity`, so it is just a matter of reading it when building our visual legend entries (thus building RGBA colors)."*

Reply:
yep, that's the re-read/edit part! haha... i just kind of put it all down... your suggestions are great. will replace.

Reply:
(you can also tell i don't know what i'm talking about LOL)

but yes, please feel free to comment on any part and i'll incorporate tomorrow.

i also removed one of the steps (step-2) as it was really uneeded, redundant. but all those changes will be clearer tomorrow